### PR TITLE
React for role

### PR DESCRIPTION
**Pull request by Unknownlocal**
`_________________ ERROR collecting tests/test_ReactForRole.py __________________
tests/test_ReactForRole.py:24: in <module>
    from cogs import ReactForRole
cogs/ReactForRole.py:25: in <module>
    UNICODE_DISCORD_EMOJI_REGEXP: re.Pattern = re.compile("^:(\w+):$")
E   AttributeError: module 're' has no attribute 'Pattern'`
in travis. Unknown why it occurs

### CODE REVIEW

**Comment by 3eak** [comment]
@Unknownlocal @JayDwee From doing some research online it (your travis problem) looks like it might be a problem with Python 3.6 - I'll put a change in to update to 3.7 to see if that fixes it for you

**Comment by JayDwee** [comment]
> @Unknownlocal @JayDwee From doing some research online it (your travis problem) looks like it might be a problem with Python 3.6 - I'll put a change in to update to 3.7 to see if that fixes it for you

@3eak Maybe change it to python 3.8 as that is what we are meant to be developing with, thanks!

File: CHANGELOG.md
Comment (JayDwee) [inline CHANGELOG.md:5]: Add to the CHANGELOG all new commands and functionality (see v0.1.0 for reference)
Also make sure you update from master again, TextFilter stuff should be here

File: KoalaBot.py
Comment (JayDwee) [inline KoalaBot.py:130]: Why was return true removed here? I haven't checked the recent usage but from the docstrings it seems pretty key
```suggestion
    return True
```

File: cogs/ColourRole.py
Comment (JayDwee) [inline cogs/ColourRole.py:282]: Were these issues before? or was this accidental. If it was a previous issue please lmk, and add it to CHANGELOG

File: cogs/Verification.py
Comment (JayDwee) [inline cogs/Verification.py:374]: thanks for these comments for your errors, very helpful for the reviewer

File: requirements.txt
Comment (JayDwee) [inline requirements.txt:32]: emoji added twice?
```suggestion

```

File: tests/test_ColourRole.py
Comment (JayDwee) [inline tests/test_ColourRole.py:288]: In future any changes like these to ColourRole should be done on the ColourRole branch so we can add any fixes like this without needing a full different branch to be ready.

File: cogs/ReactForRole.py
Comment (JayDwee) [inline cogs/ReactForRole.py:38]: maybe change default to be rfr for easier reading and have reactForRole as an alias.
Also ReactForRole doesn't work as it is the name of the whole cog
```suggestion
    @commands.group(name="rfr", aliases=["reactForRole", "react_for_role"])
```

File: cogs/ReactForRole.py
Comment (JayDwee) [inline cogs/ReactForRole.py:43]: Maybe change to `createMsg` to stay consistent with welcome and other cogs
```suggestion
    @react_for_role_group.command(name="createMsg")
```
Repeat for other "Message" commands

File: cogs/ReactForRole.py
Comment (JayDwee) [inline cogs/ReactForRole.py:44]: docstrings needed!!!

File: cogs/ReactForRole.py
Comment (JayDwee) [inline cogs/ReactForRole.py:39]: should there be docstrings here?

File: cogs/ReactForRole.py
Comment (JayDwee) [inline cogs/ReactForRole.py:30]: comments with docstrings needed everywhere, can't see any/many atm

File: cogs/ReactForRole.py
Comment (JayDwee) [reply cogs/ReactForRole.py:43]: Actually may be better as just "create"
```suggestion
    @react_for_role_group.command(name="create")
```

File: cogs/ReactForRole.py
Comment (JayDwee) [inline cogs/ReactForRole.py:485]: worth making the 60 a constant at the top so we can edit it later if required, some of these messages are quite long which makes me wonder if 120 would be better in the future. 60 is good for now, but not having this hardcoded everywhere and instead using a global/ class constant is better. Make sure to edit any 'I'll wait 60 seconds' strings to use the constant.

File: cogs/ReactForRole.py
Comment (JayDwee) [inline cogs/ReactForRole.py:51]: Recieved this error when waiting for timeout:
```
TypeError
expected string or bytes-like object```
Make sure if a timeout happens, there are no further errors passed to the user, a return when none may be ideal

**Review by JayDwee — CHANGES_REQUESTED** [review]
Not completed review, but some stuff to look at. Git seems to have done some weird stuff with your update from master tho, idk

File: cogs/ReactForRole.py
Comment (Unknownlocal) [reply cogs/ReactForRole.py:485]: Currently it's using the param as a default rather than a constant, the message request when called is automatically adjusted to whatever the arg given was.

**Comment by Unknownlocal** [comment]
> 
> 
> Not completed review, but some stuff to look at. Git seems to have done some weird stuff with your update from master tho, idk

Git has done some weird stuff with the PR, yes. I'll be manually reviewing these changes since the ColourRole changes for example shouldn't have happened

File: cogs/ColourRole.py
Comment (Unknownlocal) [reply cogs/ColourRole.py:282]: This should've been that way since before ColourRole was merged with master. Not sure when/why it's changed, but it was pre- anything to do with merging into master.

File: cogs/ReactForRole.py
Comment (Unknownlocal) [reply cogs/ReactForRole.py:39]: yes, and it's removed them for some reason. I'm gonna have to go through with docstrings everywhere again from the looks of things.

File: cogs/ReactForRole.py
Comment (Unknownlocal) [reply cogs/ReactForRole.py:79]: On travis?

**Comment by JayDwee** [comment]
Nice the weird git stuff for the pr are fixed 😃

File: cogs/ColourRole.py
Comment (JayDwee) [reply cogs/ColourRole.py:282]: Worth putting something like 'minor fixes' in the changelog so if for some strange reason people say it's breaking we at least know something for colour role was changed in this update

File: cogs/ReactForRole.py
Comment (JayDwee) [reply cogs/ReactForRole.py:79]: No, through manual testing, if you look at the bottom of #jack-test-bot you'll see the interaction

File: cogs/ReactForRole.py
Comment (JayDwee) [inline cogs/ReactForRole.py:475]: Nice comments so far!
docstrings here and lower pls <3
p.s. I realise you were probs just in the middle of doing it and I decide to spring another review on you, my bad XD

**Review by JayDwee — CHANGES_REQUESTED** [review]
Sorry, for the unnecessary, review. When you are ready for the next, just choose to request one from me in the upper right of git and dm me and I'll sort it out <3

File: cogs/ColourRole.py
Comment (Unknownlocal) [reply cogs/ColourRole.py:282]: k will do.

**Review by JayDwee — CHANGES_REQUESTED** [review]

File: cogs/ReactForRole.py
Comment (Unknownlocal) [reply cogs/ReactForRole.py:69]: Issue was it broke locally without them, and wouldn't display.

File: cogs/ReactForRole.py
Comment (Unknownlocal) [reply cogs/ReactForRole.py:224]: without the \\\ it refused to show a newline as a newline for some reason, so I left it at that, and without \" it broke locally. again, not sure why

File: cogs/ReactForRole.py
Comment (JayDwee) [reply cogs/ReactForRole.py:670]: Remove any old code

**Review by JayDwee — CHANGES_REQUESTED** [review]
Looks good to upload onto alpha bot, just remove that small excess code.
Make the final change, and then we can merge without another review

**Pull request merged**
